app/views/questionnaire.py

```diff
@@ -208,14 +212,22 @@ def get_thank_you(eq_id, form_type):  # pylint: disable=unused-argument
 
     if session_data.submitted_time:
         schema = load_schema_from_params(session_data.eq_id, session_data.form_type)
-        metadata_context = build_metadata_context_for_survey_completed(vars(session_data))
-        thank_you_template = render_theme_template(schema.json['theme'],
-                                                   template_name='thank-you.html',
-                                                   meta=metadata_context,
-                                                   analytics_ua_id=current_app.config['EQ_UA_ID'],
-                                                   survey_id=schema.json['survey_id'],
-                                                   survey_title=TemplateRenderer.safe_content(schema.json['title']))
-        return thank_you_template
+        metadata_context = build_metadata_context_for_survey_completed(session_data)
+
+        view_submission_url, view_submission_duration = None, 0
+        if is_view_submitted_response_enabled(schema.json):
+            view_submission_duration = schema.json['view_submitted_response']['duration']
+            if _is_submission_viewable(schema.json, session_data.submitted_time):
```

Comment:
Duration is only needed if the submission is viewable, so it would be better to move it to the point we work out the url. The logic for converting it to minutes should also move here.

Reply:
I was using duration as the check for showing either the link or the timeout panel on the thank you page. If it was greater than 0 then it showed the view submission content. Instead i'm now passing a boolean of *is_view_submission_enabled*